adapters/andalucia.js

```diff
@@ -0,0 +1,925 @@
+const request = baseRequest.defaults({timeout: REQUEST_TIMEOUT});
+
+exports.name = 'andalucia';
+
+export function fetchData (source, callback) {
+  let baseUrl = source.url;
+  // see if site is available. if so, build tasks, then run them in parallel
+  request(baseUrl, (err, res, body) => {
+    if (err || res.statusCode !== 200) {
+      return callback({message: 'Failed to load entry point url'});
+    }
+    // get provinces from form on source.url.
+    // use them to make taskObjs with name/id.
+    // this is helpful for descriptive error logs and using the coordinates csv
+    let $ = cheerio.load(body);
+    let tasks = [];
+    $('select[name=PROVINCIA]').children().filter((i, el) => {
+      tasks.push({
+        provinceName: $(el).text(),
+        provinceID: $(el)['0'].attribs.value
+      });
+    });
+    const now = moment();
+    // generate list of tasks
+    tasks = tasks.map((task) => {
+      let url = source.sourceURL + `${now.format('MMM').toLowerCase() + now.format('YY')}/n${task.provinceID + now.format('YYMMDD')}.htm`;
+      return generateTasks(url, task, now);
+    });
+    // execute in parallel
+    async.parallel(
+      tasks,
+      (err, results) => {
+        if (err) {
+          return callback(null, err);
+        }
+        results = _.flattenDeep(results);
+        return callback(null, results);
+      }
+    );
+  });
+}
+//
+const generateTasks = (url, task, now) => {
+  return (done) => {
+    // Try getting data from today and yesterday
+      // in both cases, if unsuccessful, return blank list
+      // when successful, return list of parsed data.
+    async.parallel([
+      (cb) => {
+        request.get(
+          url, (err, res, body) => {
+            if (err || res.statusCode !== 200) {
+              console.log(
+                'Records for ' + moment().format('DD-MM-YY') +
+                ' not reached.'
+              );
+              return cb(null, []);
+            }
+            let $ = cheerio.load(body);
+            const data = formatData($, stations);
+            return cb(null, data);
+          }
+        );
+      },
+      (cb) => {
+        url = url.replace(now.format('YYMMDD'), moment().add(-1, 'days').format('YYMMDD'));
+        request.get(
+          url, (err, res, body) => {
+            if (err || res.statusCode !== 200) {
+              console.log(
+                'Records for ' + moment().add(-1, 'days').format('DD-MM-YY') +
+                'not reached.'
+              );
+              return cb(null, []);
+            }
+            let $ = cheerio.load(body);
+            const data = formatData($, stations);
+            cb(null, data);
+          }
+        );
+      }], (err, res) => {
+      if (err) {
+        return done(null, []);
+      }
+      res = res[0].concat(res[1]);
+      done(null, res);
+    });
+  };
+};
+
+const formatData = ($, stations) => {
+  const tables = [];
+  // the tables we want don't have cellpadding 5.
```

Comment:
Agreed it is less than awesome way - the HTML provides zero id/class to latch onto to make selections, unfortunately. 

Maybe instead of relying on HTML attributes I can use the station name (since I have it in the station constant) and select the table that includes the station name + the next table.

That would get us the 2 table groups we need.